Assets/Src/Auth/REST/IHttpClientRequestAdapter.cs

```diff
@@ -2,11 +2,11 @@
 using System.Net.Http;
 using System.Threading.Tasks;
 
-namespace Src.AuthController.REST
+namespace Src.Auth.REST
 {
     public interface IHttpClientRequestAdapter
     {
         public void Request<T>(HttpMethod requestMethodType, string uri,
-            Dictionary<string, string> requestParams, TaskCompletionSource<T> tcs);
+            IEnumerable<KeyValuePair<string, string>> requestParams, TaskCompletionSource<T> tcs);
```

Comment:
Turning this method into async would be more handy than giving it a TaskCompletionSource.